golang/btcspv/utils.go

```diff
@@ -14,10 +14,47 @@ func strip0xPrefix(s string) string {
 	return s
 }
 
+// Decodes a hex string into a byte array
```

Comment:
For Go, every exported function should have a comment that starts with the function name

`// DecodeIfHex decodes a hex string into a byte array`